sdk/servicebus/service-bus/src/core/linkEntity.ts

```diff
@@ -224,35 +267,34 @@ export class LinkEntity {
   }
 
   /**
-   * Closes the Sender|Receiver link and it's underlying session and also removes it from the
-   * internal map.
+   * Closes the internally held rhea link, stops the token renewal timer and sets
+   * the this._link field to undefined.
    *
-   * @param {Sender | Receiver} [link] The Sender or Receiver link that needs to be closed and
-   * removed.
+   * @param mode Indicates the original caller.
+   * - "permanently" closes the link permanently, setting _wasClosedByUser to true which
+   * prevents it from being reinitializing.
+   * - "linkonly" closes the link but does not permanently close the LinkEntity. It can be reinitialized.
    */
-  protected async _closeLink(link?: AwaitableSender | Receiver): Promise<void> {
+  protected async closeLink(mode: "permanently" | "linkonly"): Promise<void> {
+    if (mode === "permanently") {
+      this._wasClosedByUser = true;
+    }
+
     clearTimeout(this._tokenRenewalTimer as NodeJS.Timer);
-    if (link) {
+    this._tokenRenewalTimer = undefined;
+
+    if (this._link) {
       try {
         // This should take care of closing the link and it's underlying session. This should also
         // remove them from the internal map.
+        const link = this._link;
+        this._link = undefined;
+
         await link.close();
-        log.link(
-          "[%s] %s '%s' with address '%s' closed.",
-          this._context.namespace.connectionId,
-          this._type,
```

Comment:
We are missing the link type in the logs. Was this intentional? Sender vs Receiver vs Management link would be nice to log. Can we add it to the logPrefix?

Reply:
That's a good point. I could do the "pass me your logger" trick like I've done in other spots so users can still filter properly via `DEBUG`. I'll fix that.

Reply:
I fixed this by adding in some known prefixes  (s: sender,  ms: message session, sr: streaming receiver, br: batching receiver, m: managementlink).

I also fixed the issue where I was using the generic log.link logger instead of the appropriate one.